azure/services/subnets/subnets.go

```diff
@@ -63,25 +63,16 @@ func (s *Service) Reconcile(ctx context.Context) error {
 			return errors.Wrapf(err, "failed to get subnet %s", subnetSpec.Name)
 		case err == nil:
 			// subnet already exists, update the spec and skip creation
-			var subnet infrav1.SubnetSpec
-			if subnetSpec.Role == infrav1.SubnetControlPlane {
-				subnet = s.Scope.ControlPlaneSubnet()
-			} else if subnetSpec.Role == infrav1.SubnetNode {
-				subnet = s.Scope.NodeSubnet()
-			} else {
-				continue
-			}
-
+			subnet := s.Scope.Subnet(subnetSpec.Name)
 			subnet.Role = subnetSpec.Role
-			subnet.Name = existingSubnet.Name
 			subnet.CIDRBlocks = existingSubnet.CIDRBlocks
 			subnet.ID = existingSubnet.ID
+			s.Scope.SetSubnet(subnetSpec.Name, subnet)
```

Comment:
same bug here: subnet wasn't getting updated since SubnetSpec was changed to not be a pointer in #1144